test/test_extension.py

```diff
@@ -20,13 +20,17 @@ def __init__(self, filename, buildername):
         super().__init__(filename)
         self._buildername = buildername
 
+    def valid(self):
+        return 'extension' in self.options.get('test', ['extension'])
+
     def _get_suffix(self):
         return 'txt' if self._buildername == 'text' else self._buildername
 
     def _sphinx_build(self, srcdir):
         outdir = os.path.join(srcdir, self._buildername)
         doctreedir = os.path.join(srcdir, 'doctrees')
         confdir = testenv.testdir
+        confoverrides = self.get_conf_overrides()
```

Comment:
This threw me off because the commit message doesn't explain this.

Reply:
Fair. I'll add a note about that.

Reply:
Fixed! I actually split it into its own commit, as it should have been.